services/rfq/relayer/limiter/limiter.go

```diff
@@ -0,0 +1,162 @@
+	IsAllowed(ctx context.Context, request reldb.QuoteRequest) (bool, error)
+}
+
+type limiterImpl struct {
+	client     client.EVM
+	metrics    metrics.Handler
+	quoter     quoter.Quoter
+	cfg        relconfig.Config
+	tokenNames map[string]relconfig.TokenConfig
+}
+
+// NewRateLimiter creates a new Limiter.
+// TODO: implement the sliding window: queue up requests and process them in order if cumulative volume is above limit.
+func NewRateLimiter(
+	cfg relconfig.Config,
+	q quoter.Quoter,
+	client client.EVM,
+	metricHandler metrics.Handler,
+	tokens map[string]relconfig.TokenConfig,
+) Limiter {
+	return &limiterImpl{
+		// Limiter:    ratelimit.New(cfg.MaxRFQSize),
+		client:     client,
+		metrics:    metricHandler,
+		quoter:     q,
+		cfg:        cfg,
+		tokenNames: tokens,
+	}
+}
+
+// IsAllowed returns true if the request is allowed, false otherwise.
+func (l *limiterImpl) IsAllowed(ctx context.Context, request reldb.QuoteRequest) (_ bool, err error) {
+	ctx, span := l.metrics.Tracer().Start(ctx, "limiter.IsAllowed", trace.WithAttributes(
+		attribute.Int64("block_number", int64(request.BlockNumber)),
+		attribute.Int64("origin_token_decimals", int64(request.OriginTokenDecimals)),
+		attribute.Int64("dest_token_decimals", int64(request.DestTokenDecimals)),
+		attribute.String("transaction_id", fmt.Sprintf("%x", request.TransactionID)),
+		attribute.String("sender", request.Sender.Hex()),
+		attribute.String("origin_chain_id", fmt.Sprint(request.Transaction.OriginChainId)),
+		attribute.String("origin_token", request.Transaction.OriginToken.Hex()),
+		attribute.String("origin_amount", request.Transaction.OriginAmount.String()),
+		attribute.String("status", request.Status.String()),
+		attribute.String("origin_tx_hash", request.OriginTxHash.Hex()),
+		attribute.Int64("relay_nonce", int64(request.RelayNonce)),
+	))
+
+	defer func() {
+		metrics.EndSpanWithErr(span, err)
+	}()
+
+	// if enough confirmations, allow because reorgs are rare at this point
+	hasEnoughConfirmations, err := l.hasEnoughConfirmations(ctx, request)
+	if err != nil {
+		return false, fmt.Errorf("could not check confirmations: %w", err)
+	}
+	if hasEnoughConfirmations {
+		return true, nil
+	}
+
+	// if not enough confirmations, check volume
+	withinSize, err := l.withinSizeLimit(ctx, request)
+	if err != nil {
+		return false, fmt.Errorf("could not check volume limit: %w", err)
+	}
+
+	return withinSize, nil
+}
+
+func (l *limiterImpl) hasEnoughConfirmations(ctx context.Context, request reldb.QuoteRequest) (_ bool, err error) {
+	ctx, span := l.metrics.Tracer().Start(ctx, "limiter.hasEnoughConfirmations")
+	defer func() {
+		metrics.EndSpanWithErr(span, err)
+	}()
+
+	currentBlockNumber, err := l.client.BlockNumber(ctx)
+	if err != nil {
+		return false, fmt.Errorf("could not get block number: %w", err)
+	}
+
+	requiredConfirmations, err := l.cfg.GetConfirmations(int(request.Transaction.OriginChainId))
```

Comment:
I don't think Confirmations here means what you think it does. 

https://github.com/synapsecns/sanguine/blame/2842a3adea8317f2448862b2902c6d57aba9c73b/services/rfq/relayer/relconfig/getters.go#L205 

This refers to number of confirmations for *any* tx, not foro txes over rate limit

Reply:
should I add another field to the config specifically for rate limit confirmations then?